### PR TITLE
Checking for openshift yaml recursively in the current directory

### PR DESCRIPTION
**Pull request by rupalibehera**
checking recursively for openshift.yml and removed the hardcoded path for openshift.yaml

https://github.com/fabric8io/fabric8-pipeline-library/issues/13

cc: @rawlingsj

### CODE REVIEW

File: src/io/fabric8/Fabric8Commands.groovy [hunk 1]
```diff
@@ -496,8 +496,12 @@ def getServiceURL(String serviceName, String namespace = null, String protocol =
 }
 
 def isOpenShiftS2I() {
-    def openshiftYaml = "target/classes/META-INF/fabric8/openshift.yml"
     try {
+        new File('.').eachFileRecurse(FILES) {
```
Comment (rawlingsj) [inline src/io/fabric8/Fabric8Commands.groovy]: where does FILES come from?  I think that needs to be set?

File: src/io/fabric8/Fabric8Commands.groovy [hunk 2]
```diff
@@ -496,8 +496,12 @@ def getServiceURL(String serviceName, String namespace = null, String protocol =
...
+          if(it.name.endsWith('openshift.yml')) {
+          def openshiftYaml = it.absolutePath
```
Comment (rawlingsj) [inline src/io/fabric8/Fabric8Commands.groovy]: The `def openshiftYaml` here means the rest of the function cant see it.  You can define `openshiftYaml` variable at the top of the method.

**Comment by rawlingsj** [comment]
ok thanks @rupalibehera!  I added a couple of comments.  

FWIW I tested the changes using this inline jenkins pipeline script, you can copy / paste it youself into a new Jenkins pipeline job created via the jenkins UI..

```
@Library('github.com/rupalibehera/fabric8-pipeline-library@issueid-13')
def dummy
mavenNode {
  ws{
      git 'https://github.com/rawlingsj/t3d.git'
      container(name: 'maven') {
          sh 'mvn clean install'
          def flow = new io.fabric8.Fabric8Commands()
          def rs = flow.isOpenShiftS2I()
          echo "${rs}"
      }
  }
}

```

File: src/io/fabric8/Fabric8Commands.groovy (hunk 1 above)
Comment (rupalibehera) [inline src/io/fabric8/Fabric8Commands.groovy]: Yes my bad, I missed the import statement `import static groovy.io.FileType.FILES`. I kinda did a mock testing where I had imported it.

**Comment by rupalibehera** [comment]
@rawlingsj , I have tested the above PR as suggested by you and it works fine now, let me know if you have anymore reviews on this ?

**Comment by rawlingsj** [comment]
Could you rebase this PR please @rupalibehera and we'll get it merged in

File: src/io/fabric8/Fabric8Commands.groovy [hunk 3]
```diff
@@ -498,10 +498,10 @@ def getServiceURL(String serviceName, String namespace = null, String protocol =
 }
 
 def isOpenShiftS2I() {
-    def openshiftYaml = "target/classes/META-INF/fabric8/openshift.yml"
+  def openshiftYaml = findFiles(glob: '**/openshift.yml')
```
Comment (jimmidyson) [inline src/io/fabric8/Fabric8Commands.groovy]: What about if this finds multiple files? Should they all be checked or just the first one as you're doing below?

Comment (rupalibehera) [inline src/io/fabric8/Fabric8Commands.groovy]: I am not sure if we will have multiple files, as we can see above in the diff we were earlier just checking for one file in the same location now we are just checking for it recursively in the same directory.

**Comment by rupalibehera** [comment]
@rawlingsj , I have rebased the PR

**Comment by rupalibehera** [comment]
@rawlingsj , can we merge this one ?

**Comment by rawlingsj** [comment]
[merge]

**Pull request merged**